pkg/controller/daemon/daemon_controller.go

```diff
@@ -646,6 +646,12 @@ func (dsc *DaemonSetsController) deletePod(obj interface{}) {
 	glog.V(4).Infof("Pod %s deleted.", pod.Name)
 	dsc.expectations.DeletionObserved(dsKey)
 	dsc.enqueueDaemonSet(ds)
+
+	// delete or update ds
+	if ds.Status.NumberMisscheduled > 0 && len(pod.Spec.NodeName) != 0 {
```

Comment:
`ds` is the controller of the deleted pod (scheduled pod), so we shouldn't depend on its `.Status.NumberMisscheduled`.

Reply:
#49488 requeues DaemonSet when non-daemon pods are deleted. The issue #58868 happens because we never requeue when daemon pods are deleted.

Reply:
>ds is the controller of the deleted pod (scheduled pod), so we shouldn't depend on its .Status.NumberMisscheduled.

Any reason? I only hope the resource's change can be caught, i think this 'ds' can be used because this change is caused by it.

>#49488 requeues DaemonSet when non-daemon pods are deleted. The issue #58868 happens because we never requeue when daemon pods are deleted.

yes, so i need try to requeue when daemon pods are deleted.

Reply:
Take the example you put in #58868. Let's say we have 2 DaemonSet, `A` and `B`. `A` can create pods (normal), `B` can't, because there's not enough resource on nodes. 

After `A`'s pod get deleted, we enter `deletePod`. `A`'s `.Status.NumberMisscheduled > 0` means that there's at least one pod from A scheduled on a node where that pod isn't expected to be there. Will that be `true` in this example? Why should we depend on it?